anesthetic/samples.py

```diff
@@ -611,6 +611,38 @@ def dlogX(self, nsamples=None):
         else:
             return WeightedDataFrame(dlogX, self.index, weights=self.weights)
 
+    def importance_reweighting(self, logL_new, replace=True, add=False):
+        """Perform importance re-weighting on the log-likelihood.
+
+        Parameters
+        ----------
+        logL_new: np.array
+            New log-likelihood values. Should have the same shape as `logL`.
+
+        replace: bool
+            Replace the current `logL` with the new `logL_new`.
+
+        add: bool
+            Add the new `logL_new` to the current `logL`.
+
+        Returns
+        -------
+        samples: NestedSamples
+            Importance re-weighted samples.
+        """
+        samples = merge_nested_samples((self, ))
+        if not replace ^ add:
+            raise ValueError("One and only one of args 'replace' or 'add' "
+                             "should be set to True.")
+        elif replace:
+            samples.logL = logL_new
+        elif add:
+            samples.logL += logL_new
+        samples = merge_nested_samples(
+            (samples[samples.logL > samples.logL_birth], )
+        )
```

Comment:
This use of `merge_nested_samples` on a single array to resort and recompute the live points is a really cool idiom, which is spoilt by the name of the function. Is it worth considering a re-name here to reflect this additional use (at the very least, the docstring of `merge_nested_samples` should be updated to 'one or more', and perhaps have a sentence saying that this should also be used to recompute a masked run. Alternatively, one could add a `recalculate_nested_sampling` (or equivalently named) method to NestedSamples, which is then used by merge_nested_samples.

Reply:
Good ideas, happy to integrate this into this PR, but ambivalent as to which route to take.

Reply:
I would go with the `NestedSamples.recalculate` option.

Reply:
Hmm, on closer inspection, the reason I like using `merge_nested_samples` in these situations is it's call to `pandas.concat` which returns a new pandas dataframe which subsequently gets turned into a `NestedSamples` object, thus making a deep copy.

A `NestedSamples.recalculate` method would essentially be an exact copy of `merge_nested_samples` and would not be worth calling from `merge_nested_samples`, so I'm going with a docstring update after all.

Reply:
done in [fc52995](https://github.com/williamjameshandley/anesthetic/pull/122/commits/fc529951dc7550cc5bf2fdc3de54d9eeb908ce03).

Reply:
> OK, we'll go with recommending the 'merge a single nested sampling run' idiom for now. You should still update the docstring so that it reads "Merge one or more nested sampling runs."

I did that (see preceding post with commit link)... Or do you mean somewhere else?

Reply:
Top of the docstring still reads "Merge two or more nested sampling runs." on both my clone and on github.

Reply:
Apologies, I was looking at the sentence below the `runs` parameter docstring, not the title docstring.

Fixed accordingly.